### PR TITLE
Implement [CBlock|CBlockIndex]::GetBaseVersion()

### PR DESCRIPTION
**Pull request by patricklodder**
- nVersion & 0xff to easily compare versions without aux data
- change implementations checking nVersion throughout main.ccp

### CODE REVIEW

File: src/core.h [hunk 1]
```diff
@@ -396,6 +396,12 @@ class CBlockHeader
         return nVersion / BLOCK_VERSION_CHAIN_START;
     }
 
+    // base block version without auxpow chain
+    int GetBaseVersion() const
+    {
+        return nVersion & 0x0000ffff;
```
Comment (rnicoll) [inline src/core.h]: Sorry, should have caught this earlier - aren't bits 0x0000ff00 used as flags (i.e. to indicate if the block has AuxPoW or not)? So this mask should be 0x000000ff instead?

Comment (patricklodder) [inline src/core.h]: well it's `0x62 * (0x01 << 16) | baseVersion` => `0x62 * 0x00010000 | baseVersion` => `0x00620000 | baseVersion`

Comment (rnicoll) [inline src/core.h]: I'm looking at the flag https://github.com/dogecoin/dogecoin/blob/1.9-dev/src/core.h#L33 - I think we need to filter that byte out too.

Comment (patricklodder) [inline src/core.h]: ugh you're right. rebasing

**Comment by patricklodder** [comment]
changed mask to 0xff

**Comment by rnicoll** [comment]
Thanks!

**Pull request merged**